WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModelTest.kt

```diff
@@ -0,0 +1,215 @@
+package org.wordpress.android.ui.posts
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.util.LocaleManagerWrapper
+import org.wordpress.android.viewmodel.ResourceProvider
+import java.util.Calendar
+
+class EditPostPublishSettingsViewModelTest : BaseUnitTest() {
+    @Mock lateinit var resourceProvider: ResourceProvider
+    @Mock lateinit var postSettingsUtils: PostSettingsUtils
+    @Mock lateinit var localeManagerWrapper: LocaleManagerWrapper
+    private lateinit var viewModel: EditPostPublishSettingsViewModel
+
+    private val dateCreated = "2019-05-05T14:33:20+0200"
+    private val currentCalendar = Calendar.getInstance()
+    private val dateLabel = "Updated date"
+
+    @Before
+    fun setUp() {
+        viewModel = EditPostPublishSettingsViewModel(resourceProvider, postSettingsUtils, localeManagerWrapper)
+        currentCalendar.set(2019, 6, 6, 10, 20)
+        whenever(localeManagerWrapper.getCurrentCalendar()).thenReturn(currentCalendar)
+        whenever(postSettingsUtils.getPublishDateLabel(any())).thenReturn(dateLabel)
+    }
+
+    @Test
+    fun `on start sets values and buils formatted label`() {
+        val post = PostModel()
+        post.dateCreated = dateCreated
+
+        val expectedLabel = "Scheduled for 2019"
+        whenever(postSettingsUtils.getPublishDateLabel(post)).thenReturn(expectedLabel)
+        var label: String? = null
+        viewModel.onPublishedLabelChanged.observeForever {
+            label = it
+        }
+
+        viewModel.start(post)
+
+        assertThat(viewModel.year).isEqualTo(2019)
+        assertThat(viewModel.month).isEqualTo(4)
+        assertThat(viewModel.day).isEqualTo(5)
+        assertThat(viewModel.hour).isEqualTo(14)
```

Comment:
This check fails for me for some reason -> `Expected :14 Actual   :5`

Reply:
hm, there might be a timezone issue, I'll look into it 👍